library/src/main/java/okhttp3/mock/MockInterceptor.java

```diff
@@ -101,6 +101,7 @@ public MockInterceptor behavior(Behavior behavior) {
     @Override
     public Response intercept(Chain chain) throws IOException {
         Request request = chain.request();
+        RequestCache.INSTANCE.setRequest(request);
```

Comment:
This won't scale because:
1) Parallel request will interfere each other and override the instance
2) The latest request will be leaked on `RequestCache` preventing it from being garbage collected.

I think `RequestBody` has a "caching" capability. You should try to use it or replace given the body with a catchable one.